File: 0560-subarray-sum-equals-k/0560-subarray-sum-equals-k.java
class Solution {
    public int subarraySum(int[] nums, int k) {
        int n=nums.length;
        if(n==0){
            return 0;
        }
        int count=0;
        int prefixsum=0;
        Map<Integer, Integer> map=new HashMap<>();
        map.put(0,1);
        for (int i=0;i<n;i++){
            prefixsum+=nums[i];
            int diff=prefixsum-k;
            if(map.containsKey(diff)){
                count+=map.get(diff);
            }
            map.put(prefixsum,map.getOrDefault(prefixsum,0)+1);
        }
        return count;
    }
}

File: 0560-subarray-sum-equals-k/0560-subarray-sum-equals-k.py
class Solution:
    def subarraySum(self, nums: List[int], k: int) -> int:
        count=0
        prefixsum=0
        map={0:1}
        for num in nums:
            prefixsum+=num
            diff=prefixsum-k
            if diff in map:
                count+=map.get(diff)
            map[prefixsum]=map.get(prefixsum,0)+1
        return count
    
        